### PR TITLE
Updated packages after porting xenopsd from oasis to jbuilder.

### PR DESCRIPTION
**Pull request by kc284**
This depends on https://github.com/xapi-project/xenopsd/pull/421

### CODE REVIEW

File: packages/xs-extra/xapi-xenopsd.master/opam [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
-        ["./setup.bin" "-uninstall"]
-        ["ocamlfind" "remove" "xenopsd"]
-]
+build: [[ "jbuilder" "build" "-p" name "-j" jobs ]]
+build-test: [[ "jbuilder" "runtest" "-p" name "-j" jobs ]]
 depends: [
-  "oasis" {build}
-  "ocamlfind" {build}
+  "jbuilder" {build}
+  "base"
```
Comment (lindig) [inline packages/xs-extra/xapi-xenopsd.master/opam]: Why do we need base? Had we missed this previously? Is this the JaneStreet Base library?

Comment (edwintorok) [reply packages/xs-extra/xapi-xenopsd.master/opam]: I think this initially came in as a dependency of `configurator`, but then we've seen its impossible to install base without upgrading the other janestreet packages, so it should probably be removed for now.

Comment (kc284) [reply packages/xs-extra/xapi-xenopsd.master/opam]: Yes, apologies, I forgot to remove it!

**Comment by mseri** [comment]
I get a failure on the build:

```
[ERROR] The compilation of xapi failed at "jbuilder build -p xapi".

#=== ERROR while installing xapi.master =======================================#

# opam-version         1.2.2 (aa258ecc06d3aea5a67f442a4ffd23f2a457180b)

# os                   linux

# command              jbuilder build -p xapi

# path                 /home/opam/.opam/4.04.2/build/xapi.master

# compiler             4.04.2

# exit-code            1

# env-file             /home/opam/.opam/4.04.2/build/xapi.master/xapi-471-5b4ccf.env

# stdout-file          /home/opam/.opam/4.04.2/build/xapi.master/xapi-471-5b4ccf.out

# stderr-file          /home/opam/.opam/4.04.2/build/xapi.master/xapi-471-5b4ccf.err

### stderr ###

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

# Error: External library "xenopsd" not found.

# -> required by "ocaml/xapi/jbuild (context default)"

# Hint: try: jbuilder external-lib-deps --missing -p xapi @install

#=== ERROR while installing xapi-xenopsd-simulator.master =====================#

# opam-version         1.2.2 (aa258ecc06d3aea5a67f442a4ffd23f2a457180b)

# os                   linux

# command              jbuilder build -p xapi-xenopsd-simulator -j 4

# path                 /home/opam/.opam/4.04.2/build/xapi-xenopsd-simulator.master

# compiler             4.04.2

# exit-code            1

# env-file             /home/opam/.opam/4.04.2/build/xapi-xenopsd-simulator.master/xapi-xenopsd-simulator-471-f5a5f5.env

# stdout-file          /home/opam/.opam/4.04.2/build/xapi-xenopsd-simulator.master/xapi-xenopsd-simulator-471-f5a5f5.out

# stderr-file          /home/opam/.opam/4.04.2/build/xapi-xenopsd-simulator.master/xapi-xenopsd-simulator-471-f5a5f5.err

### stderr ###

# [...]

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

# Error: External library "xenopsd" not found.

# -> required by "test/jbuild (context default)"

# Hint: try: jbuilder external-lib-deps --missing -p xapi-xenopsd-simulator @install

#=== ERROR while installing xapi-xenopsd-xc.master ============================#

# opam-version         1.2.2 (aa258ecc06d3aea5a67f442a4ffd23f2a457180b)

# os                   linux

# command              jbuilder build -p xapi-xenopsd-xc -j 4

# path                 /home/opam/.opam/4.04.2/build/xapi-xenopsd-xc.master

# compiler             4.04.2

# exit-code            1

# env-file             /home/opam/.opam/4.04.2/build/xapi-xenopsd-xc.master/xapi-xenopsd-xc-471-e875bf.env

# stdout-file          /home/opam/.opam/4.04.2/build/xapi-xenopsd-xc.master/xapi-xenopsd-xc-471-e875bf.out

# stderr-file          /home/opam/.opam/4.04.2/build/xapi-xenopsd-xc.master/xapi-xenopsd-xc-471-e875bf.err

### stderr ###

# [...]

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

# Error: External library "xapi-xenopsd" is unavailable.

# -> required by "xl/jbuild (context default)"

# External library "xapi-xenopsd" is not available because it depends on the following libraries that are not available:

# - c_stubs -> not found

# Hint: try: jbuilder external-lib-deps --missing -p xapi-xenopsd-xc @install

#=== ERROR while installing xapi-xenopsd-xenlight.master ======================#

# opam-version         1.2.2 (aa258ecc06d3aea5a67f442a4ffd23f2a457180b)

# os                   linux

# command              jbuilder build -p xapi-xenopsd-xenlight -j 4

# path                 /home/opam/.opam/4.04.2/build/xapi-xenopsd-xenlight.master

# compiler             4.04.2

# exit-code            1

# env-file             /home/opam/.opam/4.04.2/build/xapi-xenopsd-xenlight.master/xapi-xenopsd-xenlight-471-7e1a45.env

# stdout-file          /home/opam/.opam/4.04.2/build/xapi-xenopsd-xenlight.master/xapi-xenopsd-xenlight-471-7e1a45.out

# stderr-file          /home/opam/.opam/4.04.2/build/xapi-xenopsd-xenlight.master/xapi-xenopsd-xenlight-471-7e1a45.err

### stderr ###

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

#        ocaml (internal)

# findlib: [WARNING] Interface topdirs.cmi occurs in several directories: /home/opam/.opam/4.04.2/lib/ocaml, /home/opam/.opam/4.04.2/lib/ocaml/compiler-libs

# No rule found for _build/default/config.ml
```

**Comment by mseri** [comment]
I think there is some further jbuild/opam file cleanup needed

**Comment by edwintorok** [comment]
I think I've seen failures like this when I tried to refer to a private library from a public library, I just had to make sure that everything I link to has a `public_name`, otherwise projects outside of original build dir couldn't refer to it.

**Comment by mseri** [comment]
You should be able to refer to them as long as you are building, public_name is only when the private library needs to be installed because it is used by something else

**Comment by edwintorok** [comment]
Yes I was able to refer to them from the same project, btu when another project tried to use it, it failed to link, I'm guessing `c_stubs` is the name of an internal library here from xenopsd.

**Comment by mseri** [comment]
`xapi` still needs fixing, the simulator is looking for `xenopsd` as well and the spec file needs fixing

**Comment by mseri** [comment]
```
#=== ERROR while installing xapi.master =======================================#
# opam-version         1.2.2 (aa258ecc06d3aea5a67f442a4ffd23f2a457180b)
# os                   linux
# command              jbuilder build -p xapi
# path                 /home/opam/.opam/4.04.2/build/xapi.master
# compiler             4.04.2
# exit-code            1

### stderr ###
# Error: External library "xenopsd" not found.
# -> required by "ocaml/xapi/jbuild (context default)"

# Hint: try: jbuilder external-lib-deps --missing -p xapi @install

#=== ERROR while installing xapi-xenopsd-simulator.master =====================#
# opam-version         1.2.2 (aa258ecc06d3aea5a67f442a4ffd23f2a457180b)
# os                   linux
# command              jbuilder build -p xapi-xenopsd-simulator -j 4
# path                 /home/opam/.opam/4.04.2/build/xapi-xenopsd-simulator.master
# compiler             4.04.2
# exit-code            1

### stderr ###
# Error: External library "xenopsd" not found.
# -> required by "test/jbuild (context default)"

# Hint: try: jbuilder external-lib-deps --missing -p xapi-xenopsd-simulator @install
```

**Comment by mseri** [comment]
I still think we should call the library `xenopsd` for backward compatibility and have `xapi-xenopsd` be a dummy package installing `xenopsd`.

This would have avoided half of the issues even if it is ugly. And we would have fixed all these dummy libraries all together asap.

**Comment by mseri** [comment]
The error now is independent on this. I am going to merge it

**Pull request merged**